pkg/volume/plugins.go

```diff
@@ -131,26 +131,26 @@ type VolumePluginMgr struct {
 
 // Spec is an internal representation of a volume.  All API volume types translate to Spec.
 type Spec struct {
-	Name                   string
-	VolumeSource           api.VolumeSource
-	PersistentVolumeSource api.PersistentVolumeSource
-	ReadOnly               bool
+	Name             string
+	Volume           *api.Volume
```

Comment:
spec.Name is convenient to avoid the nil checks everywhere the caller wants the name.  There are enough references to it in Kubelet's volumes.go and all the plugins (in builders and cleaners) that it's a nice convenience.